src/tests/domains/StarknetDomainHost.t.sol

```diff
@@ -0,0 +1,184 @@
+        assertEq(host.wards(address(this)), 1);
+        assertEq(host.live(), 1);
+    }
+
+    function testDepost() public {
+        vat.suck(address(123), address(this), 100 * RAD);
+        daiJoin.exit(address(this), 100 ether);
+        dai.approve(address(host), 100 ether);
+
+        host.setCeiling(10 ether);
+
+        uint256 l2Address = 2345;
+
+        uint256 depositAmount = 10 ether;
+        uint256 fee = 144;
+
+        uint256 balanceBefore = dai.balanceOf(address(this));
+
+        (uint256 depositAmountLow, uint256 depositAmountHigh) = split(depositAmount);
+
+        uint256[] memory payload = new uint256[](4);
+        payload[0] = l2Address;
+        payload[1] = depositAmountLow;
+        payload[2] = depositAmountHigh;
+        payload[3] = uint256(uint160(address(this)));
+
+        vm.expectEmit(true, true, true, true);
+        emit LogMessageToL2(address(host), host.guest(), DEPOSIT, payload, 0, fee);
+
+        host.deposit{value: 144}(depositAmount, l2Address);
+
+        assertEq(dai.balanceOf(address(this)), balanceBefore - depositAmount);
+
+    }
+
+    function testAddressValidation() public {
+        uint256 amount = 1;
+
+        vm.expectRevert('StarknetDomainHost/invalid-address');
+        host.deposit(amount, 0);
+
+        vm.expectRevert('StarknetDomainHost/invalid-address');
+        host.deposit(amount, SN_PRIME);
+
+        vm.expectRevert('StarknetDomainHost/invalid-address');
+        host.deposit(amount, type(uint256).max);
+
+        vm.expectRevert('StarknetDomainHost/invalid-address');
+        host.deposit(amount, L2_DAI);
+    }
+
+    function testCeilingTooLow() public {
+        vat.suck(address(123), address(this), 100 * RAD);
+        daiJoin.exit(address(this), 100 ether);
+        dai.approve(address(host), 100 ether);
+
+        host.setCeiling(1 ether);
+
+        uint256 l2Address = 2345;
+
+        vm.expectRevert('StarknetDomainHost/above-ceiling');
+        host.deposit(10 ether, l2Address);
+    }
+
+    function testSetCeiling() public {
+        host.setCeiling(12 ether);
+        assertEq(host.ceiling(), 12 ether);
+    }
+
+    function testUnauthorizedSetCeiling() public {
+        host.deny(address(this));
+        vm.expectRevert('DomainHost/not-authorized');
+        host.setCeiling(12 ether);
+    }
+
+    function testDepositTooLarge() public {
+        host.setCeiling(10 ether);
+        host.setMaxDeposit(1 ether);
+
+        uint256 l2Address = 2345;
+
+        vm.expectRevert('StarknetDomainHost/above-max-deposit');
+        host.deposit(2 ether, l2Address);
+    }
+
+    function testSetMaxDeposit() public {
+        host.setMaxDeposit(12 ether);
+        assertEq(host.maxDeposit(), 12 ether);
+    }
+
+    function testUnauthorizedSetMaxDeposit() public {
+        host.deny(address(this));
+        vm.expectRevert('DomainHost/not-authorized');
+        host.setMaxDeposit(12 ether);
+    }
+
+}
+
+function split(uint256 value) pure returns (uint256, uint256) {
```

Comment:
can't this be imported from `StarknetDomainHost` ?

Reply:
Not sure if it deduplication is worth the resulting extension of StarknetDomainHost public interface.